### ISSUE TITLE
iCloud Sync Error for select folders

### ISSUE THREAD

**Issue by skyesofrock** [nylas/nylas-mail#3253]
##### Are there any related issues?

Not that I can see

##### What operating system are you using?
Sierra 10.12.3

##### What version of Nylas Mail are you using?
10.0.20

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?
No

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
It looks like mostly an iCloud issue, but my G Suite trash folder is also struggling to sync.

##### Is the issue reproducible with a particular attachment, message, signature, etc?

Not that I'm aware of.

##### Description
Basically a few folders from my iCloud email are not syncing at all (along with the trash folder from my G Suite). Running Mail with Debug Flags shows only one error, and the popup saying "Encountered an error while syncing" is permanently in place. The error reads:  

`/Applications/Nylas Mail.app/Contents/Resources/app.asar/node_modules/fbjs/lib/warning.js:36 Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `SyncActivity`. See https://fb.me/react-warning-keys for more information.`  

There's no correlation between the folders that would indicate it's an account issue, so I can't work out why this isn't working. I also have 2FA enabled, using an app password if that helps at all.

**Comment by entreprenik2** [comment]
Maybe related...

I keep getting **"Encountered an error while syncing `<email@me.com>`"**. Oddly enough, I have two iCloud accounts - one an older one that I still use with `@me.com` and a newer one. The older one is the one getting this error.

Nothing in Developer Console when running Nylas Mail with Debug Flags apart from a possible memory leak in EventEmitter.

These are the `/var/folders/xq/v2lxchv50db8x79s70rywzyh0000gn/T/Nylas-Mail-7138.7145.log` logs:

```{"host":"Nikitas-MacBook-Air.local","timestamp":"2017-02-05T08:22:41.585Z","payload":["ActionBridge: default Action Bridge Broadcasting: willMakeAPIRequest"]}
{"host":"Nikitas-MacBook-Air.local","timestamp":"2017-02-05T08:22:42.234Z","payload":[]}
{"host":"Nikitas-MacBook-Air.local","timestamp":"2017-02-05T08:22:42.558Z","payload":["ActionBridge: default Action Bridge Broadcasting: willMakeAPIRequest"]}
{"host":"Nikitas-MacBook-Air.local","timestamp":"2017-02-05T08:22:45.304Z","payload":["ActionBridge: default Action Bridge Broadcasting: notifyPluginsChanged"]}
{"host":"Nikitas-MacBook-Air.local","timestamp":"2017-02-05T08:22:46.632Z","payload":["ActionBridge: default Action Bridge Broadcasting: didMakeAPIRequest"]}
{"host":"Nikitas-MacBook-Air.local","timestamp":"2017-02-05T08:22:46.830Z","payload":["ActionBridge: default Action Bridge Broadcasting: didMakeAPIRequest"]}
{"host":"Nikitas-MacBook-Air.local","timestamp":"2017-02-05T08:22:49.815Z","payload":["ActionBridge: default Action Bridge Received: db-store-trigger"]}
(last one repeats indefinitely)```

**Comment by jamiemcconnell** [comment]
I have the same iCloud issue as @skyesofrock 
Folders have synced (i.e. i can see the folders names) but nothing inside any of them.
Tried to pull debug logs but i don't see any actual errors 👎

**Comment by jstejada** [comment]
thanks for the details!I think the error you are seeing must be related to this one https://github.com/nylas/nylas-mail/issues/3198#issuecomment-275507500

**Comment by entreprenik2** [comment]
Just resolved the iCloud sync issue by going into iCloud.com and deleting some old folders (Mailbox, Mailpilot, Airmail). My hunch is that nested folders (Mailbox/Foo) might have been an issue...

**Comment by jstejada** [comment]
this should be fixed on 1.0.26 which was released today! Thanks for your patience!

**Comment by skyesofrock** [comment]
Can confirm it's working, thanks for the fix!
